### PR TITLE
new blog post

### PR DESCRIPTION
**Pull request by calcsam**
New blog post with perf testing!

### CODE REVIEW

File: docs/blog/2018-06-24-investigating-gatsby-build-performance-at-scale/index.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+The three most memory-intensive operations that scale with site size are **processing markdown**, **creating pages**, and **processing images with gatsby-image**.
+
+# Benchmark Parameters
+
+_Site tests_
+
+In order to measure the “build performance cost” of each of these three factors, we’ve benchmarked Gatsby build times for two different sites, one image-heavy and one markdown-page heavy.
+
+The first site is FreeCodeCamp’s Guide, one of the largest open-source Gatsby sites with around 3,000 pages. The second site is an official example, using-gatsby-image, copied into its own repo. We’ve run two different versions of this: first, with around 25 images and second, with around 250 images. Both sites are running on Gatsby v2.
```
Comment (m-allanson) [inline docs/blog/2018-06-24-investigating-gatsby-build-performance-at-scale/index.md]: Link to FreeCodeCamp and the Gatsby image example?

File: docs/blog/2018-06-24-investigating-gatsby-build-performance-at-scale/index.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+Tests were run on three different machines; on a local development machine (2015 Macbook Pro), on an AWS t2 micro free tier instance, and an AWS memory-optimized m4 instance.
+
+_Cold start vs warm start_
+
+Tests were run both with (“warm start) and without (“cold start”) a pre-existing, local Gatsby cache.
+
+In most cases when code changes and all cases when content changes, Gatsby will preserve the cache from your previous build, giving you a warm start. The cache is only invalidated in rare instances, for example when the plugin changes.
+
+Whether you’re able to access the cache from previous builds depends on your CI setup, but many CI providers such as Netlify offer the option of preserving the previous build cache.
```
Comment (m-allanson) [inline docs/blog/2018-06-24-investigating-gatsby-build-performance-at-scale/index.md]: Configuring Netlify's cache directory is (I think) an undocumented feature, maybe drop _such as Netlify_?

**Comment by calcsam** [comment]
Thanks @m-allanson! Addressed the previous comments.

**Comment by m-allanson** [comment]
@calcsam I just merged https://github.com/gatsbyjs/gatsby/pull/6152, so will wait until tomorrow before merging this.

**Comment by m-allanson** [comment]
Cherry picked to `master`

**Pull request merged**